### PR TITLE
Fixes regression which checks wrong db_backend on centos

### PR DESCRIPTION
**Pull request by b10n1k**
to fix regression by https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12463

- Verification run: https://openqa.suse.de/tests/5955362

### CODE REVIEW

File: lib/containers/common.pm [hunk 1]
```diff
@@ -265,7 +265,7 @@ sub test_rpm_db_backend {
     my ($running_version, $sp, $host_distri) = get_os_release("$runtime run $image");
     record_info('RPM check', "The rpm db backend in the image $image is $backend");
     # TW and SLE 15-SP3+ uses rpm-ndb in the image
-    if ($host_distri eq 'opensuse-tumbleweed' || ($host_distri eq 'sles' && check_version('>=15-SP3', "$running_version-SP$sp", qr/\d{2}(?:-sp\d)?/))) {
+    if ($host_distri eq 'opensuse-tumbleweed' || ($host_distri =~ /(sles|centos)/ && check_version('>=15-SP3', "$running_version-SP$sp", qr/\d{2}(?:-sp\d)?/))) {
         $expected = 'ndb';
     }
     die("The rpm db backend should be $expected") if ($backend ne $expected);
```
Comment (jlausuch) [inline lib/containers/common.pm:271]: maybe we even need to change this condition as well... cause centos is throwing some other extra character
https://openqa.suse.de/tests/5955362#step/podman_image/1

Comment (jlausuch) [reply lib/containers/common.pm:271]:
```suggestion
    die("The rpm db backend should be $expected") if ($expected !~ m/$backend/);
```

can you try this?

**Comment by jlausuch** [comment]
I think the approach is wrong. We don't care if the host is centos or not. What matters is the container image. In this case, it's sle 15.3 what we are testing, which should have ndb, but for some reason it returns 
```
WRpU3
ndb
```
So, it would be sufficient to fix this by my suggestion `if ($expected !~ m/$backend/);`

**Comment by b10n1k** [comment]
no needed

**Pull request closed without merging**